sljit_src/sljitProtExecAllocator.c

```diff
@@ -83,22 +83,21 @@ struct chunk_header {
        as it only uses local variables
 */
 
+#if !(defined(__NetBSD__) && defined(MAP_REMAPDUP))
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
 
 #ifndef O_NOATIME
 #define O_NOATIME 0
 #endif
 
-#ifdef __O_TMPFILE
+/* this is a linux extension available since kernel 3.11 */
 #ifndef O_TMPFILE
-#define O_TMPFILE	(__O_TMPFILE | O_DIRECTORY)
-#endif
+#define O_TMPFILE 020200000
```

Comment:
This is something new. Basically we try to pass this flag, and see what happens? I am ok with it, just want to make sure I understand correctly. These flags are harmless on other systems, don't they?

Reply:
that is valid only in Linux. and wouldn't be needed if the code was compiled with -D_GNU_SOURCE as it will come included from fcntl.h

in NetBSD will just return an error because it is a directory and continue with the fallback